### PR TITLE
MVP for `vibes-patch`

### PR DESCRIPTION
**Pull request by bmourad01**
This PR brings the minimal functionality for the `vibes-patch` tool (and library). Highlights:

- Unlike the old patcher, we only take a single patch specification at a time. However, we will inform the user at what address the patch was placed and how big the patch was in bytes so that, if we want to fit multiple patches into the same space, we will know where to start for the next patch.
- The code is more target-agnostic now. We can use a first-class module that has all the target-specific functionality (since assemblers for different architectures have different quirks).

### CODE REVIEW

File: vibes-tools/tools/vibes-constants/lib/asm.mli [hunk 1]
```diff
@@ -1,5 +1,4 @@
 (** Predefined labels for use with the system assembler. *)
 
-val relative_patch_placement : string
```
Comment (philzook58) [inline vibes-tools/tools/vibes-constants/lib/asm.mli:3]: We don' need this anymore?

Comment (bmourad01) [reply vibes-tools/tools/vibes-constants/lib/asm.mli:3]: I didn't see it being used anywhere. Was there a reason for adding it?

Comment (philzook58) [reply vibes-tools/tools/vibes-constants/lib/asm.mli:3]: There probably was a reason, but if it's gone, good riddance. I bet it was used to calculate jumps in the assembler, but there could easily be a different method that's better

**Review by jtpaasch — APPROVED** [review]
👍

**Pull request merged**